packages/sdk/src/types/runtime/assets/asset.ts

```diff
@@ -82,12 +84,51 @@ export abstract class Asset implements AssetLike {
 	/** @inheritdoc */
 	public get source() { return this._source; }
 
-	protected constructor(public manager: AssetManager, def: Partial<AssetLike>) {
+	/** @inheritdoc */
+	public get prefab(): Prefab { return null; }
+	/** @inheritdoc */
+	public get mesh(): Mesh { return null; }
+	/** @inheritdoc */
+	public get material(): Material { return null; }
+	/** @inheritdoc */
+	public get texture(): Texture { return null; }
+	/** @inheritdoc */
+	public get sound(): Sound { return null; }
+
+	/** A promise that resolves when the asset is finished loading */
+	public get created() { return this._loadedPromise; }

```

Comment:
>created() [](start = 12, length = 9)

Model this after the way it's done in Actor. That implementation handles the load failure case, where future calls to created() will reject if the load failed.

Reply:
Oh wait, your's does too, just a different way. Deleting comment.

---
In reply to: [302370180](https://github.com/microsoft/mixed-reality-extension-sdk/pull/384#discussion_r302370180) [](ancestors = 302370180)